internal/service/ecs/service.go

```diff
@@ -464,6 +489,8 @@ func resourceServiceCreate(d *schema.ResourceData, meta interface{}) error {
 		input.TaskDefinition = aws.String(v.(string))
 	}
 
+	input.DeploymentConfiguration = &ecs.DeploymentConfiguration{}
```

Comment:
I moved this line up as it was clearing all properties that were set.. (see line 481 in the diff)

Reply:
Ideally we refactor this to properly expand and flatten the DeploymentConfiguration, but it's all over the place with conditionals at the moment. Something we can tackle in the future.